script.rb

```diff
@@ -0,0 +1,144 @@
+# rubocop:disable Style/CaseEquality, Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity, Metrics/MethodLength, Metrics/ModuleLength
+module Enumerable
+  def my_each
+    i = 0
+    while i < length
+      return to_enum unless block_given?
+
+      yield(self[i])
+      i += 1
+    end
+  end
+
+  def my_each_with_index
```

Comment:
* [ ] should have working `my_each_with_index`  identical and returns the same thing as ruby's [`each_with_index`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-each_with_index). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_each_with_index.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

  ```return to_enum(my_each_with_index) unless block_given?```
 instead of
 ```return to_enum unless block_given?```
that way
```[1,2,3].my_each_with_index``` ```<Enumerator: [1, 2, 3]:my_each_with_index>```
```[1,2,3].each_with_index``` =>  ```<Enumerator: [1, 2, 3]:each_with_index>```

Reply:
method my_each_with_index updated